torch/tensor.py

```diff
@@ -274,7 +274,7 @@ def repeat(self, *sizes):
             raise ValueError('Number of dimensions of repeat dims can not be '
                              'smaller than number of dimensions of tensor')
 
-        xtensor = src.new().set_(src)
```

Comment:
Hi,

I don't think you want to clone here, this is not needed,
A minimal fix would be to change the `resize_` call l284 by a `view` (and remove line 290 that also contains a `resize_` and that is redundant with l284).

Reply:
I may have to read the cpp sources to figure out what these do.